pkg/scheduler/framework/v1alpha1/interface.go

```diff
@@ -111,10 +112,10 @@ func (s *Status) Code() Code {
 
 // Message returns message of the Status.
 func (s *Status) Message() string {
-	if s == nil {
+	if s == nil || s.IsSuccess() {
 		return ""
 	}
-	return s.message
+	return s.AsError().Error()
```

Comment:
I believe we need to return the list of reasons/errors so that we can iterate over them when generating the histogram: https://github.com/kubernetes/kubernetes/blob/a73e0f2112d285e4872037428dff8dda55229039/pkg/scheduler/core/generic_scheduler.go#L99

I think we should have two functions, one is ```Message```  which concatenates all reasons, and one named ```Reasons``` which return the slice. ```Reasons``` is used when generating the histogram, and ```Message`` used for logs and all other purposes.